std/src/main/scala/algebra/std/array.scala

```diff
@@ -0,0 +1,42 @@
+package algebra
+package std
+
+import scala.{specialized ⇒ sp}
+
+package object array extends ArrayInstances
+
+trait ArrayInstances {
+  implicit def arrayEq[@sp A: Eq]: Eq[Array[A]] = new ArrayEq[A]
```

Comment:
I'm fine with such a change. I don't care (or particularly like) squashed commits. I prefer to keep the whole history.